src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -13,8 +9,80 @@ internal partial class LocalRewriter
     {
         public override BoundNode VisitTupleCreationExpression(BoundTupleCreationExpression node)
         {
-            var rewrittenArguments = VisitList(node.Arguments);
-            return new BoundObjectCreationExpression(node.Syntax, node.ConstructorOpt, rewrittenArguments);
+            ImmutableArray<BoundExpression> rewrittenArguments = VisitList(node.Arguments);
+            return RewriteTupleCreationExpression(node, rewrittenArguments);
+        }
+
+        /// <summary>
+        /// Converts the expression for creating a tuple instance into an expression creating a ValueTuple (if short) or nested ValueTuples (if longer).
+        ///
+        /// For instance, for a long tuple we'll generate:
+        /// creationExpression(ctor=largestCtor, args=firstArgs+(nested creationExpression for remainder, with smaller ctor and next few args))
+        /// </summary>
+        private BoundNode RewriteTupleCreationExpression(BoundTupleCreationExpression node, ImmutableArray<BoundExpression> rewrittenArguments)
+        {
+            int remainder;
+            int chainLength = TupleTypeSymbol.NumberOfValueTuples(rewrittenArguments.Length, out remainder);
+
+            NamedTypeSymbol underlyingTupleType = ((TupleTypeSymbol)node.Type).UnderlyingTupleType;
+            ArrayBuilder<NamedTypeSymbol> underlyingTupleTypeChain = GetUnderlyingTypeChain(underlyingTupleType);
+
+            var currentCreation = default(BoundObjectCreationExpression);
+
+            {
+                // make a creation expression for the smallest type
+                ImmutableArray<BoundExpression> ctorArguments = ImmutableArray.Create(rewrittenArguments, (chainLength - 1) * TupleTypeSymbol.chunk7, remainder);
+
+                var ctor = (MethodSymbol)Binder.GetWellKnownTypeMember(_compilation, TupleTypeSymbol.GetTupleCtor(remainder), _diagnostics, syntax: node.Syntax);
+                if ((object)ctor == null)
+                {
+                    underlyingTupleTypeChain.Free();
```

Comment:
Sorry, I should have given more detail previously. The recommendation is to avoid ```try/finally``` in cases where it is not necessary to dispose the resource (e.g.: return a builder to the pool) when an `Exception` is thrown _and_ where removing the `finally` does not introduce unnecessary complexity to the code (e.g.: multiple exit points that require disposing).